ui/src/app/pages/admin/workspace/admin-workspace.tsx

```diff
@@ -36,166 +35,156 @@ interface Props
-        {workspace && (
-          <div>
-            {activeStatus === WorkspaceActiveStatus.ACTIVE && (
-              <AdminLockWorkspace
-                {...{ workspace }}
-                reload={async () =>
-                  await this.populateFederatedWorkspaceInformation()
-                }
-              />
-            )}
-            <BasicInformation
-              {...{ workspace, activeStatus }}
-              reload={async () => await this.populateWorkspaceDetails()}
+  const populateFederatedWorkspaceInformation = async () => {
+    const { ns } = props.match.params;
+    setLoadingWorkspace(true);
+
+    try {
+      const newCloudStorageTraffic =
+        await workspaceAdminApi().getCloudStorageTraffic(ns);
+      setCloudStorageTraffic(newCloudStorageTraffic);
+    } catch (error) {
+      console.log('Error loading cloud storage traffic: ', error);
+    }
+
+    try {
+      const newRuntimes = await workspaceAdminApi().adminListRuntimes(ns);
+      setRuntimes(newRuntimes);
+    } catch (error) {
+      handleDataLoadError(error);
+    }
+
+    try {
+      const newUserApps =
+        await workspaceAdminApi().adminListUserAppsInWorkspace(ns);
+      setUserApps(newUserApps);
+    } catch (error) {
+      handleDataLoadError(error);
+    }
+
+    try {
+      const newWorkspaceDetails =
+        await workspaceAdminApi().getWorkspaceAdminView(ns);
+      setWorkspaceDetails(newWorkspaceDetails);
+    } catch (error) {
+      handleDataLoadError(error);
+    } finally {
+      setLoadingWorkspace(false);
+    }
+  };
+
+  useEffect(() => {
+    props.hideSpinner();
+    populateFederatedWorkspaceInformation();
+  }, []);
+
+  useEffect(() => {
+    populateFederatedWorkspaceInformation();
+  }, [props.match.params.ns]);
+
+  const populateWorkspaceDetails = async () => {
+    const { ns } = props.match.params;
+    setLoadingWorkspace(true);
+
+    try {
+      const newWorkspaceDetails =
+        await workspaceAdminApi().getWorkspaceAdminView(ns);
+      setWorkspaceDetails(newWorkspaceDetails);
+    } catch (error) {
+      handleDataLoadError(error);
+    } finally {
+      setLoadingWorkspace(false);
+    }
+  };
+
+  const { profile } = profileStore.get();
+  const { collaborators, resources, workspace, activeStatus } =
+    workspaceDetails || {};
+  const { workspaceObjects, cloudStorage } = resources || {};
+  const { researchPurpose } = workspace || {};
+
+  return (
+    <div style={{ margin: '1.5rem' }}>
+      {dataLoadError && (
+        <ErrorDiv>
+          Error loading data. Please refresh the page or contact the development
+          team.
+        </ErrorDiv>
+      )}
+      {loadingWorkspace && <SpinnerOverlay />}
+      {workspace && (
+        <div>
+          {activeStatus === WorkspaceActiveStatus.ACTIVE && (
+            <AdminLockWorkspace
+              {...{ workspace }}
+              reload={async () => await populateFederatedWorkspaceInformation()}
```

Comment:
can simplify this and similar calls to `reload={populateFederatedWorkspaceInformation}`:
* `() => foo()` is equivalent to `async() => await foo()` 
* we no longer need to use `() => {}` for "this-safety"